### PR TITLE
feat(@vtmn/svelte): add `VtmnIcon` component

### PR DESCRIPTION
**Pull request by Tlahey**
Add new component VtmnIcon on svelte

- Rework of `rollup.conf.js` for svelte (to include guidelines)
- Add function `objectToStyle` to convert an object to a string for the style attribute
- Add storybook for `VtmnIcon`
- Add some missing configuration on svelte storybook (lib + story sort)

### CODE REVIEW

**Review by GaspardMathon — APPROVED** [review]

File: packages/showcases/svelte/package.json [hunk 1]
```diff
@@ -12,8 +12,12 @@
   "devDependencies": {
     "@babel/core": "^7.17.0",
     "@storybook/addon-a11y": "6.3.13",
+    "@storybook/addon-actions": "6.3.13",
+    "@storybook/addon-backgrounds": "6.3.13",
+    "@storybook/addon-docs": "6.3.13",
     "@storybook/addon-essentials": "6.3.13",
     "@storybook/addon-svelte-csf": "^1.1.0",
+    "@storybook/addon-viewport": "6.3.13",
```
Comment (lauthieb) [inline packages/showcases/svelte/package.json:20]: Same remark as mentioned in `main.js` file

File: packages/showcases/svelte/.storybook/main.js [hunk 2]
```diff
@@ -4,9 +4,13 @@ module.exports = {
...
   addons: [
-    '@storybook/addon-svelte-csf',
-    '@storybook/addon-essentials',
     '@storybook/addon-a11y',
+    '@storybook/addon-actions',
+    '@storybook/addon-backgrounds',
+    '@storybook/addon-docs',
+    '@storybook/addon-essentials',
+    '@storybook/addon-svelte-csf',
+    '@storybook/addon-viewport',
```
Comment (lauthieb) [inline packages/showcases/svelte/.storybook/main.js:13]: Addons "actions", "backgrounds", "docs" & "viewport" are already present in "essentials" so you don't need to add them.

CleanShot 2022-03-06 at 22 51 21

File: packages/sources/svelte/rollup.config.js [hunk 3]
```diff
@@ -4,60 +4,64 @@ import svelte from 'rollup-plugin-svelte';
...
+  ({ folder, components }) =>
+    components.map((component) => ({
+      input: `src/${baseDir}/${folder}/${component}/${component}.svelte`,
+      ...svelteOptions(component),
+    }));
+
+export default [
+  ...src.components.flatMap(predicate('components')),
+  ...src.guidelines.flatMap(predicate('guidelines')),
+];
```
Comment (lauthieb) [inline packages/sources/svelte/rollup.config.js:67]: Are you sure there are no breaking changes for current users?

File: packages/sources/svelte/src/guidelines/iconography/VtmnIcon/enums.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export const VARIANT = {
+  DEFAULT: 'default',
+  BRAND: 'brand',
+  REVERSED: 'reversed',
+  POSITIVE: 'positive',
+  INFORMATION: 'information',
+  WARNING: 'warning',
+  NEGATIVE: 'negative',
+};
```
Comment (lauthieb) [inline packages/sources/svelte/src/guidelines/iconography/VtmnIcon/enums.js:9]: Can you please rename this const by "VtmnIconVariant" as we did for badge & tag components enums? Thanks :)

**Review by lauthieb — CHANGES_REQUESTED** [review]

File: packages/showcases/svelte/.storybook/main.js (hunk 2 above)
Comment (Tlahey) [reply packages/showcases/svelte/.storybook/main.js:13]: Yes !
In storybook css showcase contains `@storybook/addon-docs` and others with `@storybook/addon-essentials` it needs to be replace too ?

File: packages/sources/svelte/rollup.config.js (hunk 3 above)
Comment (Tlahey) [reply packages/sources/svelte/rollup.config.js:67]: On my tests there is no problem, the flatMap will transform the 2D arrays into 1 and we concat the 2 arrays into 1.

File: packages/sources/svelte/src/guidelines/iconography/VtmnIcon/enums.js (hunk 4 above)
Comment (Tlahey) [reply packages/sources/svelte/src/guidelines/iconography/VtmnIcon/enums.js:9]: I prefer that the const exports are in uppercase, it's ok if I rename by `VTMN_ICON_VARIANT` ?

**Review by GaspardMathon — DISMISSED** [review]

File: packages/sources/svelte/src/guidelines/iconography/VtmnIcon/enums.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export const VARIANT = {
+  DEFAULT: 'default',
+  BRAND: 'brand',
```
Comment (lauthieb) [inline packages/sources/svelte/src/guidelines/iconography/VtmnIcon/enums.js:3]:
```suggestion
  BRAND_PRIMARY: 'brand-primary',
```

File: packages/sources/svelte/src/guidelines/iconography/VtmnIcon/VtmnIcon.svelte [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+  let className;
+  /**
+   * @type {string} Custom classes to apply to the component.
+   */
+  export { className as class };
+
+  const retrieveSemanticColor = (variant) => {
+    switch (variant) {
+      case VARIANT.BRAND:
```
Comment (lauthieb) [inline packages/sources/svelte/src/guidelines/iconography/VtmnIcon/VtmnIcon.svelte:29]:
```suggestion
      case VARIANT.BRAND_PRIMARY:
```

**Review by lauthieb — CHANGES_REQUESTED** [review]

File: packages/sources/svelte/src/guidelines/iconography/VtmnIcon/enums.js (hunk 4 above)
Comment (lauthieb) [reply packages/sources/svelte/src/guidelines/iconography/VtmnIcon/enums.js:9]: As discussed together, feel free to propose the format `VTMN_ICON_VARIANT` but please do the same for other existing enums in `@vtmn/svelte` stack :)

File: packages/showcases/svelte/.storybook/main.js (hunk 2 above)
Comment (lauthieb) [reply packages/showcases/svelte/.storybook/main.js]: Yes! Feel free to add `@storybook/addon-essentials` in the CSS showcase and replacing all packages "backgrounds" etc.

Comment (Tlahey) [reply packages/showcases/svelte/.storybook/main.js]: Done :)

**Comment by lauthieb** [comment]
Well done! Thanks <3

**Pull request merged**